bee-api/bee-rest-api/src/endpoints/routes/api/v2/milestone_by_milestone_index.rs

```diff
@@ -30,24 +30,27 @@ pub(crate) fn filter<B: StorageBackend>(
 ) -> BoxedFilter<(impl Reply,)> {
     self::path()
         .and(warp::get())
-        .and(has_permission(ROUTE_MILESTONE, public_routes, allowed_ips))
+        .and(has_permission(
+            ROUTE_MILESTONE_BY_MILESTONE_INDEX,
+            public_routes,
+            allowed_ips,
+        ))
         .and(with_tangle(tangle))
-        .and_then(|milestone_index, tangle| async move { milestone(milestone_index, tangle) })
+        .and_then(|milestone_index, tangle| async move { milestone_by_milestone_index(milestone_index, tangle) })
         .boxed()
 }
 
-pub(crate) fn milestone<B: StorageBackend>(
+pub(crate) fn milestone_by_milestone_index<B: StorageBackend>(
     milestone_index: MilestoneIndex,
     tangle: ResourceHandle<Tangle<B>>,
 ) -> Result<impl Reply, Rejection> {
-    match tangle.get_milestone_metadata(milestone_index) {
-        Some(milestone) => Ok(warp::reply::json(&MilestoneResponse {
-            milestone_index: *milestone_index,
-            message_id: milestone.message_id().to_string(),
-            timestamp: milestone.timestamp(),
-        })),
-        None => Err(reject::custom(CustomRejection::NotFound(
-            "cannot find milestone".to_string(),
-        ))),
+    let milestone_id = match tangle.get_milestone_metadata(milestone_index) {
+        Some(milestone_metadata) => *milestone_metadata.milestone_id(),
+        None => return Err(reject::custom(CustomRejection::NotFound("data not found".to_string()))),
```

Comment:
Does it need to be this error message? If not it would be clearer if it's `milestone not found`

Reply:
Overall I don't really like the error handling of this crate, it needs a refactoring at some point.

Reply:
Agree. The Axum PR already introduces some improvements. So let's keep it for now as it is IMO.